charts/shoot-core/components/charts/apiserver-proxy/templates/apiserver-proxy-pod-mutatingwebhook.yaml

```diff
@@ -0,0 +1,45 @@
+apiVersion: {{ include "webhookadmissionregistration" . }}
+kind: MutatingWebhookConfiguration
+metadata:
+  name: apiserver-proxy.networking.gardener.cloud
+  annotations:
+    networking.gardener.cloud/description: |-
+      This webhook adds `KUBERNETES_SERVICE_HOST` environment variable
+      to all containers and init containers matched by it.
+  labels:
+    gardener.cloud/role: system-component
+    origin: gardener
+webhooks:
+- admissionReviewVersions:
+  - v1beta1
+  clientConfig:
+    caBundle: {{ .Values.webhook.caBundle }}
+    url: https://127.0.0.1:9443/webhook/pod-apiserver-env
+  failurePolicy: Ignore
+  matchPolicy: Exact
+  name: apiserver-proxy.networking.gardener.cloud
+  namespaceSelector:
+    matchExpressions:
+    - key: apiserver-proxy.networking.gardener.cloud/inject
+      operator: NotIn
+      values:
+      - disable
+  objectSelector:
+    matchExpressions:
+    - key: apiserver-proxy.networking.gardener.cloud/inject
+      operator: NotIn
+      values:
+      - disable
+  reinvocationPolicy: Never
+  rules:
+  - apiGroups:
+    - ""
+    apiVersions:
+    - v1
+    operations:
+    - CREATE
```

Comment:
I'm not so sure about this question, but I want to double-check: Should `UPDATE` be added here?
It's probably a super rare use case to change one of the few updatable fields in a Pod with some update mechanism other than patch. Nevertheless, this could cause the API server to return an error indicating, that `env` cannot be changed, although the user didn't change anything. So this could _potentially_ break some users, who do such weird stuff, right?

Reply:
https://github.com/kubernetes/kubernetes/blob/a86afc12df023e8a45a1c3b6523aed38c3e6cd18/pkg/apis/core/validation/validation.go#L3927-L3930

You cannot update this field.

Reply:
Yes, exactly. And that's why an update request with the env var set will fail. 
If the webhook would also add it on updates, the field won't be changed and the request succeeds.
As said, this is a pretty rare use case. But nevertheless, should we consider it?

Reply:
> As said, this is a pretty rare use case. But nevertheless, should we consider it?

The request would be invalid in the first place (you'll need to get resource to perform such operation), so I guess not.

Reply:
Yeah, that's also right. It was just a minor question anyways, it's not too relevant.
But I guess, it wouldn't hurt to add `UPDATE` as well (so the env vars will always be ensured, no matter if `CREATE` or `UPDATE`), would it?